File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/cpp/F10.P1.cpp
#include <cstdio>

int narrow(long long big) {
// BEGIN PROBE F10.P1
    int small = static_cast<int>(big);
    return small;
// END PROBE F10.P1
}

int main() { std::printf("%d\n", narrow(2147483648LL)); }

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/rust/F10.P1.rs
use std::convert::TryFrom;

fn narrow(big: i64) -> i32 {
// BEGIN PROBE F10.P1
let small = i32::try_from(big).unwrap_or(0);
small
// END PROBE F10.P1
}

fn main() {
    println!("{}", narrow(2147483648));
}
